sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -169,7 +169,8 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
     }
 
     private def checkHintNonEquiJoin(hint: JoinHint): Unit = {
-      if (hintToShuffleHashJoin(hint) || hintToSortMergeJoin(hint)) {
+      if (hintToShuffleHashJoin(hint) || hintToPreferShuffleHashJoin(hint) ||
+        hintToSortMergeJoin(hint)) {
```

Comment:
`PreferShuffleHashJoin` is an internal hint so user cann't use it, we don't need do this check

Reply:
Yeah I am aware of this hint is internal only. I am more thinking towards to catch and expose the bug where we apply `PREFER_SHUFFLE_HASH` internally by mistake (e.g. for the added unit test query in `JoinHintSuite.scala`). I feel it's no harm to add here and help catch more bugs in the future. But if you guys think we should not add it, I can also remove it. cc @cloud-fan .